src/flow_solver/flow_solver_cases/gaussian_bump.cpp

```diff
@@ -19,41 +20,64 @@ GaussianBump<dim, nstate>::GaussianBump(const PHiLiP::Parameters::AllParameters
 template <int dim, int nstate>
 std::shared_ptr<Triangulation> GaussianBump<dim,nstate>::generate_grid() const 
 {
-    std::shared_ptr <Triangulation> grid = std::make_shared<Triangulation>(
+    if constexpr(dim==2) {
+        std::shared_ptr <Triangulation> grid = std::make_shared<Triangulation>(
         this->mpi_communicator,
         typename dealii::Triangulation<dim>::MeshSmoothing(
             dealii::Triangulation<dim>::smoothing_on_refinement |
             dealii::Triangulation<dim>::smoothing_on_coarsening));
+        const unsigned int number_of_refinements = this->all_param.flow_solver_param.number_of_mesh_refinements;
+        const double channel_length = this->all_param.flow_solver_param.channel_length;
+        const double channel_height = this->all_param.flow_solver_param.channel_height;
+        const double bump_height = this->all_param.flow_solver_param.bump_height;
 
-    const unsigned int number_of_refinements = this->all_param.flow_solver_param.number_of_mesh_refinements;
-    const double channel_length = this->all_param.flow_solver_param.channel_length;
-    const double channel_height = this->all_param.flow_solver_param.channel_height;
-    const double bump_height = this->all_param.flow_solver_param.bump_height;
+        std::vector<unsigned int> n_subdivisions(dim);
+        n_subdivisions[0] = this->all_param.flow_solver_param.number_of_subdivisions_in_x_direction;
+        n_subdivisions[1] = this->all_param.flow_solver_param.number_of_subdivisions_in_y_direction;
+        // n_subdivisions[2] = this->all_param.flow_solver_param.grid.gaussian_bump.number_of_subdivisions_in_z_direction;
```

Comment:
Removed in the next commit. It was kept in case anyone wanted to extend the internal grid generator to 3D for the gaussian bump, which only works for the 2D case in the current implementation. For 3D case, the current workflow is to use gmsh_reader.